flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/FileStorePathFactory.java

```diff
@@ -133,6 +133,9 @@ public Long latestSnapshotId() {
             FileStatus[] statuses = fs.listStatus(snapshotDir);
 
             long latestId = Snapshot.FIRST_SNAPSHOT_ID - 1;
+            if (statuses == null) {
+                return null;
+            }
```

Comment:
What will trigger this? When `snapshotDir` does not exist? If that is the case shall we throw an exception stating that we cannot find the specified snapshot directory?

Reply:
I prefer to relax this restriction. read a non-existent table to return an empty collection. It is not necessary for writer to create the `snapshot` directory.

Reply:
> What will trigger this? When snapshotDir does not exist? If that is the case shall we throw an exception stating that we cannot find the specified snapshot directory?

TestAtomicRenameFileSystem#listStatus

Reply:
+1 for throw meaningful exception here

Reply:
I think we can:
- add a debug log and return 0 for `file not exists`.
- Throw exception for `listStatus return null`.

Reply:
I think we can:
- add a debug log and return 0 for `file not exists`.
- Throw exception for `listStatus return null`.